k2/csrc/fsa_algo_test.cu

```diff
@@ -149,6 +150,44 @@ TEST(ArcSort, NonEmptyFsaVec) {
   }
 }
 
+
+TEST(FsaAlgo, FixFinalLabels) {
+  // src_state dst_state label cost
+  std::string s = R"(0 1 10 -1.2
+    0 2  6 -2.2
+    0 3  9 -2.2
+    1 2  8  -3.2
+    1 3  6  -4.2
+    2 3  5 -5.2
+    2 4  4  -6.2
+    3 5 -1  -7.2
+    5
+    )";
+  for (auto &context : {GetCudaContext(), GetCpuContext()}) {
+    Fsa fsa = FsaFromString(s);
+    fsa = fsa.To(context);
+    std::vector<Fsa*> fsa_ptrs { &fsa, &fsa };
+    K2_CHECK_EQ(fsa_ptrs.size(), 2);
+    FsaVec fsas = Stack(0, 2, fsa_ptrs.data(), nullptr);
+
+
+    Array1<int32_t> labels(context, "[ 0 -1 1 2 3 4 5 0 ]"),
+        correct_labels(context, "[ 0 0 1 2 3 4 5 -1 ]");
+
+    FixFinalLabels(fsa, labels.Data(), 1);
+    EXPECT_EQ(true, Equal(labels, correct_labels));
+
+
```

Comment:
Maybe you forgot to call `FixFinalLabels` for the FsaVec `fsas`.

Also, you need to save a copy of `labels` for later reference when testsing `fsas` as it is modified in
line 177
```cpp
FixFinalLabels(fsa, labels.Data(), 1);
```